src/ValueRepresentation.js

```diff
@@ -136,6 +136,42 @@ class ValueRepresentation {
         return tag;
     }
 
+    /**
+     * Removes padding byte, if it exists, from the last value in a multiple-value data element.
+     *
+     * This function ensures that data elements with multiple values maintain their integrity for lossless
```

Comment:
I think the pad byte is an artifact of the binary encoding, and should always be stripped - that makes it consistent as to what value one gets.  Depending on the maximum character length means one ends up getting different results depending on the exact boundary/parse conditions which is a recipe for errors and bugs.